src/components/MessageList.tsx

```diff
@@ -0,0 +1,486 @@
+					onClick={store.clearParsedMessages}>
+					Clear
+				</button>
+
+				<button
+					disabled={store.getCurrentMessagesArray.length === 0}
+					className='mainButton'
+					onClick={exportFn}>
+					Export
+				</button>
+
+				<button
+					disabled={store.editMessageMode || store.getCurrentMessagesArray.length === 0}
+					className='mainButton'
+					onClick={() => {
+						setReplayMode(!isReplay);
+					}}>
+					{isReplay ? (
+						<div style={{ display: 'flex' }}>
+							<div className='spinner'></div>Stop
+						</div>
+					) : (
+						'Replay'
+					)}
+				</button>
+
+				<input
+					disabled={store.editMessageMode}
+					value=''
+					id='file'
+					type='file'
+					accept='.json'
+					onChange={e => {
+						loadFromFile(e.target.files);
+					}}
+				/>
+			</div>
+		</div>
+	);
+};
+
+const MessageEntity = (props: { message: ParsedMessageItem | ActMessageItem }) => {
+	if (isParsedMessageItem(props.message)) {
+		return (
+			<div className='messageEntity'>
+				<p>
+					<b>session: </b>
+					{props.message.sessionId}
+				</p>
+				<p>
+					<b>dictionary: </b>
+					{props.message.dictionary}
+				</p>
+				<p>
+					<b>messageType: </b>
+					{props.message.messageType}
+				</p>
+			</div>
+		);
+	}
+	if (isActMessageItem(props.message)) {
+		return (
+			<div>
+				<p>
+					<b>actBox: </b>
+					{props.message.actBox}
+				</p>
+				<p>
+					<b>fullServiceName: </b>
+					{props.message.fullServiceName}
+				</p>
+				<p>
+					<b>methodName: </b>
+					{props.message.methodName}
+				</p>
+			</div>
+		);
+	}
+	return null;
+};
+
+interface MessageItemProps extends EditMessageProps {
+	index: number;
+	message: ParsedMessageItem | ActMessageItem;
+	selectMessage: (index: number) => void;
+	setDelay: (index: number) => void;
+	deleteMessage: (index: number) => void;
+}
+
+const MessageItem = ({
+	index,
+	message,
+	indicators,
+	selectMessage,
+	setDelay,
+	deleteMessage,
+	editMessageMode,
+	editedMessageIndex,
+}: MessageItemProps) => {
+	const [delay, setDelayValue] = useState(message.delay.toString());
+	return (
+		<div className='messageCard'>
+			<div
+				onClick={() => {
+					selectMessage(index);
+				}}>
+				<MessageEntity message={message} />
+				<p>
+					<b>delay: </b>
+					{editMessageMode && editedMessageIndex === index ? (
+						<input
+							className='delayInput'
+							type='number'
+							value={delay}
+							onChange={e => {
+								setDelay(Number(e.target.value));
+								setDelayValue(e.target.value);
+							}}
+						/>
+					) : (
+						message.delay
+					)}
+					ms
+				</p>
+			</div>
+			<MessageCardControls
+				deleteMessage={deleteMessage}
+				editMessageMode={editMessageMode}
+				index={index}
+				indicator={indicators[index]}
+			/>
+		</div>
+	);
+};
+
+interface MessageCardControlsProps {
+	deleteMessage: (index: number) => void;
+	editMessageMode: boolean;
+	index: number;
+	indicator: Indicator;
+}
+
+const MessageCardControls = ({
```

Comment:
create folder message-list and split these components into 2 or more files
e.g.
MessageList,
MessageItem,